servers/lib/src/files/files.module.ts

```diff
@@ -1,9 +1,36 @@
 import { Module } from '@nestjs/common';
-import FilesResolver from './resolvers/files.resolver.js';
-import FilesServiceFactory from './services/files-service.factory.js';
-import LocalFilesService from './services/local-files.service.js';
+import FilesResolver from './files.resolver.js';
+import { GitFilesModule } from './git/git-files.module.js';
+import { LocalFilesModule } from './local/local-files.module.js';
+import LocalFilesService from './local/local-files.service.js';
+import GitFilesService from './git/git-files.service.js';
+import {
+  FILE_SERVICE,
+  IFilesService,
+} from './interfaces/files.service.interface.js';
+import FilesServiceFactory from './files-service.factory.js';
+import { ConfigService } from '@nestjs/config';
+import { CONFIG_MODE } from '@/enums/config-mode.enum.js';
 
 @Module({
-  providers: [FilesResolver, LocalFilesService, FilesServiceFactory],
+  imports: [LocalFilesModule, GitFilesModule],
+  providers: [
+    FilesResolver,
+    {
+      provide: FILE_SERVICE,
+      useFactory: (
+        configService: ConfigService,
+        localFilesService: LocalFilesService,
+        gitFilesService: GitFilesService,
+      ) => {
+        const fileServices = new Map<CONFIG_MODE, IFilesService>([
+          [localFilesService.getMode(), localFilesService],
```

Comment:
@nichlaes 
The `getMode()` function can be directly invoked in `files-service.factory.ts`. Passing an array

`const fileServices = Array<IFilesService>([localFilesService, gitFilesService],])`

is sufficient